File: Java/src/main/java/questions/ReverseLinkedList.java
package questions;


/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode(int x) { val = x; }
 * }
 */
public class ReverseLinkedList {

    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode reHead = reverseList(head.next);
        head.next.next = head;
        head.next = null;
        return reHead;
    }


}

File: Go/questions/ReverseLinkedList.go
package questions

import "github.com/leetcode/common"

func ReverseList(head *common.ListNode) *common.ListNode {
	if head == nil || head.Next == nil {
		return head
	}

	newHead := ReverseList(head.Next)
	head.Next.Next = head
	head.Next = nil
	return newHead
}
